src/mono/wasm/runtime/crypto-worker.ts

```diff
@@ -33,6 +33,27 @@ export function dotnet_browser_simple_digest_hash(ver: number, input_buffer: num
     return 1;
 }
 
+export function dotnet_browser_sign(hashAlgorithm: number, key_buffer: number, key_len: number, input_buffer: number, input_len: number, output_buffer: number, output_len: number): number {
+    mono_assert(!!mono_wasm_crypto, "subtle crypto not initialized");
+
+    const msg = {
+        func: "sign",
+        type: hashAlgorithm,
+        key: Array.from(Module.HEAPU8.subarray(key_buffer, key_buffer + key_len)),
+        data: Array.from(Module.HEAPU8.subarray(input_buffer, input_buffer + input_len))
+    };
+
+    const response = mono_wasm_crypto.channel.send_msg(JSON.stringify(msg));
+    const signResult = JSON.parse(response);
+    if (signResult.length > output_len) {
+        console.info("dotnet_browser_sign: about to throw!");
+        throw "SIGN HASH: Sign length exceeds output length: " + signResult.length + " > " + output_len;
```

Comment:
We are not really handling this anywhere. If I try to trigger this exception, then tests get stuck with:
```
  info: Starting:    System.Security.Cryptography.Tests.dll
  info: dotnet_browser_sign: about to throw!
  fail: [out of order message from the browser]: http://127.0.0.1:33501/dotnet.js 2:69923 Uncaught
```

Reply:
How are you triggering this? As previously noted, the callers should already have validated higher up in the callstack that the destination was large enough.

Reply:
I wanted to see how this exception would show up, so I manually triggered it by changing the condition to `if (signResult.length > 0)`.

Reply:
Since there is no way for a user to get into this situation, I don't think it is a concern.

https://github.com/dotnet/runtime/issues/69740 tracks adding more robust error handling to this code.